airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/test/pipeline.py

```diff
@@ -310,21 +311,39 @@ async def run_poe_tasks_for_package(
     airbyte_repo_dir = await get_filtered_airbyte_repo_dir(dagger_client, poetry_package_path)
     package_dir = await get_poetry_package_dir(airbyte_repo_dir, poetry_package_path)
     package_config = await get_airbyte_ci_package_config(package_dir)
-    container = prepare_container_for_poe_tasks(
-        dagger_client, airbyte_repo_dir, package_config, poetry_package_path, pipeline_context_params
-    )
+
     logger = logging.getLogger(str(poetry_package_path))
 
     if not package_config.poe_tasks:
         logger.warning("No poe tasks to run.")
         return []
 
-    poe_task_results = []
-    async with asyncer.create_task_group() as poe_tasks_task_group:
-        for task in package_config.poe_tasks:
-            poe_task_results.append(
-                poe_tasks_task_group.soonify(run_and_log_poe_task_results)(
-                    pipeline_context_params, str(poetry_package_path), container, task, logger
+    poe_task_results: List[asyncer.SoonValue] = []
+    return_results = []
+
+
+    if pipeline_context_params["python_version"]:
```

Comment:
If I'm reading this correctly, `pipeline_context_params["python_version"]` could be set from the `--python-version` CLI flag, then we will run only the version provided. If that's the case, `package_config.python_versions` will be set to just the single version provided in the option. Right?

Reply:
I'd argue you could simplify the code and say "you know what, no magic, it always follows pyproject.toml setting. Want a single version? Cool, just set it in pyproject.toml".

Reply:
@natikgadzhi I don't mind that per se but hear me out. We have two slightly different functional cases where it can be scheduled. The first is by a developer locally to run code in the most compatible environment. The second is CI. At GHA we need to run each version separately and we most likely want to run all versions for developers locally because that's what we require anyway. I didn't intend to use this tag manually. It's purely there to parallelize and split execution on CI.

Reply:
I agree with @natikgadzhi here.

> The first is by a developer locally to run code in the most compatible environment.

It's more convenient for developers to run poe tasks on different python version by switching poetry environment. `airbyte-ci` has significant overhead when running locally so I expect it to be mainly use in CI.

> It's purely there to parallelize and split execution on CI which you can't do at all if you set one version.

I'm not sure I get the parallelization argument. You already have concurrent poe task executions as you create the tasks inside a group using a for loop and python version

Reply:
I was considering `airbyte-ci` as a tool of final resolution. You are correct, no one would probably ever use it as a main runner for tests :D. But it's an open-source project and it doesn't have the priviledge of ppl doing "dirty commits" to cover potential diverge in behavior in reproducible airbyte-ci and local run. The infmaous "Fixed (2)" commit messages ;).

I think airbyte-ci is an amazing way to have more control over how clean you can have your PR without relying on CI that much. And as that it has more meaning doing cross-version check independently of what version you used in venv.

Reply:
@alafanechere actually wait. It does have the privilege. Sorry, ICI being free and running at your forks slept my mind. I still wonder if supporting multiple pythons is a good idea for CDK. It's not something we can containerize and set in stone, unlike component runners, right?

Reply:
I agree that `airbyte-ci` should be used locally, as a final check before pushing for instance, to catch edge cases. I think this is what you mean by "final resolution" 😄

Reply:
@alafanechere exactly!